app/processing-core.js

```diff
@@ -18,25 +33,46 @@ const balancer = (data, countWorkers, method) => {
   const size = Math.floor(len / countWorkers);
   const tasks = [];
   for (let i = 0; i < countWorkers; i++) {
-    tasks[i] = Buffer.from(data.slice(i * size, i * size + size));
+    const offset = i * size;
+    const end = i === countWorkers - 1 ? len : offset + size;
+    tasks[i] = data.slice(offset, end);
   }
 
   let finished = 0;
 
-  return new Promise((resolve) => {
+  return new Promise((resolve, reject) => {
+    const onError = (err) => {
+      removeListeners(workers, ['message', 'error'], [onMessage, onError]);
```

Comment:
Rewrite it for array of objects
```suggestion
      const events = [
        {
          name: 'message',
          listener: onMessage,
        },
        {
          name: 'error',
          listener: onError,
        },
      ]
      removeListeners(workers, ['message', 'error'], [onMessage, onError]);
```